model/dex.go

```diff
@@ -0,0 +1,43 @@
+package model
+
+import (
+	"net/url"
+)
+
+type Dex struct {
+	Enabled         bool             `yaml:"enabled"`
+	Url             string           `yaml:"url"`
+	ClientId        string           `yaml:"clientId"`
+	Username        string           `yaml:"username"`
+	Groups          string           `yaml:"groups,omitempty"`
+	CaFile          string           `yaml:"caFile,omitempty"`
```

Comment:
FYI you can implement `UnmarshalYAML` like [this](https://github.com/kubernetes-incubator/kube-aws/blob/master/core/nodepool/config/config.go#L161-L167) to automatically fill default values into a Dex struct.